### PR TITLE
feat(ui-react-native): Add theming for Button primitive

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

- Add theming
- Add accessibility role
- Update unit tests

#### Description of how you validated changes

Ran in iOS and Android. 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/primitives/Button/Button.tsx [hunk 1]
```diff
@@ -7,28 +7,37 @@ import {
...
 } from 'react-native';
 
-import { styles } from './styles';
+import { useTheme } from '../../theme';
+import { getThemedStyles } from './styles';
 import { ButtonProps } from './types';
 
 export default function Button({
+  accessibilityRole = 'button',
   children,
```
Comment (ioanabrooks) [inline packages/react-native/src/primitives/Button/Button.tsx:16]: Thanks for adding default accessibility role! We should also support `variants` and `size`  for the button.

File: packages/react-native/src/primitives/Button/__tests__/Button.spec.tsx [hunk 2]
```diff
@@ -1,24 +1,69 @@
...
-const title = 'A preesable button';
+const title = 'A pressable button';
 const Title = () => <Text>{title}</Text>;
 
 describe('Button', () => {
   it('renders as expected with a string passed as children', () => {
-    const button = TestRenderer.create(<Button>{title}</Button>);
-    expect(button.toJSON()).toMatchSnapshot();
+    const { toJSON } = render(<Button>{title}</Button>);
+    expect(toJSON()).toMatchSnapshot();
```
Comment (ioanabrooks) [inline packages/react-native/src/primitives/Button/__tests__/Button.spec.tsx:15]: Can we add some assertions about the default accessibility role and the label?

File: packages/react-native/src/primitives/Button/__tests__/Button.spec.tsx [hunk 3]
```diff
@@ -1,24 +1,69 @@
...
 
   it('renders as expected with a component passed as children', () => {
-    const button = TestRenderer.create(
+    const { toJSON } = render(
       <Button>
         <Title />
       </Button>
     );
-    expect(button.toJSON()).toMatchSnapshot();
+    expect(toJSON()).toMatchSnapshot();
```
Comment (ioanabrooks) [inline packages/react-native/src/primitives/Button/__tests__/Button.spec.tsx:24]: Any assertions we can add here?

**Review by calebpollman — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]
LGTM with the understanding that variants will be added as followup

**Pull request merged**